src/message-stream.ts

```diff
@@ -379,21 +403,4 @@ export class MessageStream extends PassThrough {
   private _setHighWaterMark(stream: PullStream): void {
     stream._readableState.highWaterMark = this._options.highWaterMark!;
   }
-  /**
-   * Promisified version of gRPCs Client#waitForReady function.
-   *
-   * @private
-   *
-   * @param {object} client The gRPC client to wait for.
-   * @returns {Promise}
-   */
-  private async _waitForClientReady(client: ClientStub): Promise<void> {
```

Comment:
I know we talked about deleting this, so I was expecting it, do we also need to enable channel pooling?

Reply:
That option is available by default. 
const pubsub = new PubSub(
  {
    'grpc_gcp.apiConfig': {
      "channelPool": {
        "maxSize": 5
        },
    }
  }
);

Reply:
Oh, this might be problematic. My understanding is that we're making the switch to the pure JS version of gRPC soon and it doesn't support gcp. @alexander-fenster can you confirm that?

Reply:
That's right. If you need to use this `grpc_gcp` functionality you'll need to do two things:
1. pass C-code gRPC as a parameter to client constructor;
2. set the `grpc_gcp` options in Pub/Sub code since they will be removed from gax.

I'm going to do the same for Spanner. If Pub/Sub is also affected, we can postpone it's upgrade to gax v1.0.0 until this is fixed in Pub/Sub code (together with Spanner).

Reply:
Thanks @alexander-fenster! Is the plan to keep Spanner on the C version until gcp support finds its way to grpc-js?

Reply:
Yes. Also, we will remove all grpc-gcp code from gax (actually, already removed, just not yet released as v1.0.0) since it's not a good place for it (and an extra dependency on grpc), so libraries that want to use grpc-gcp should depend on it directly.

Reply:
Good to know. I think we're going to want postpone the gax upgrade here unless @ajaaym can think of another fix that doesn't involve channel pooling.

Reply:
@callmehiphop WDYT of option in getClient of pubsub to provide new client vs cached one? Then depending on maxStreams we can get create maxStreams/100 + 1 client.

Reply:
I think that sounds good. Out of curiosity have we tested against grpc-js to see if it is affected by this?

Reply:
I was playing around with that yesterday and looks like its not that stable yet. Yes there is a limitation of 100 concurrent stream. Also I dont see option to set flow control limit which would be needed for load balancing ultra slow consumer. When testing with 100 streams I was randomly getting this error along with DEADLINE EXCEEDED. I think its better we hold on to upgrade until we have this option and well tested with different conditions.

```
Error [ERR_STREAM_WRITE_AFTER_END]: write after end
    at writeAfterEnd (_stream_writable.js:243:12)
    at ClientHttp2Stream.Writable.write (_stream_writable.js:291:5)
    at filterStack.sendMessage.then (/Users/ajaymovalia/work/node/testwithexpress2/node_modules/@grpc/grpc-js/build/src/call-stream.js:396:34)
Emitted 'error' event at:
    at onwriteError (_stream_writable.js:431:12)
    at onwrite (_stream_writable.js:456:5)
    at process._tickCallback (internal/process/next_tick.js:63:19)
```